codeChunks.Rmd

```diff
@@ -0,0 +1,156 @@
+---
+title: "Frequently Used Code Chunks"
+author: "Vivian Tran"
+date: "3/7/2018"
+output: github_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+These are some code chunks that I frequently come back to when processing data for the Arctic Data Center. 
+
+#Reading in raw data
+##Single data file
+```{r eval=FALSE}
+df <- read.table("path/to/data", 
+               header=T, 
+               fill = T, # blank fields are added for rows that have unequal length,
+               sep= ",", # put "," for .csv file, "\t" for files with values separated by tabs
+               na.strings = c('','NA')) # fills blank rows with NA's 
+```
+Some things to note: I specify fill=T because I often get an error related to the lengths of rows/columns:
+
+```{r echo=FALSE}
+cat("Error in scan(file = file, what = what, sep = sep, quote = quote, dec = dec,  : 
+  line 1 did not have 10 elements")
+```
+
+This usually happens in .txt files.
+
+  
+##Multiple data files 
+I use these chunks when I want to read in several data files that share the same column names and formatting. I usually group my data files into different folders according to the type of data and/or formatting to facilitate reading in the data. This will help automate reformatting later on. 
+```{r eval=FALSE}
+# grab data paths from the folder that data is stored in 
+# "path" specifies name of folder that data paths are stored 
+# full.names = T produces full paths to fukes instead of just file name
+
+rawPaths <- dir(path = "path/to/folder", full.names = T) 
+```
+
+Read in data using a for loop. Remember to initialize all variables that you will be using outside of the for loop. 
+```{r eval=FALSE}
+dataList <- vector("list", length(rawPaths)) # makes an empty list with same length as file paths vector
+i=0 
+for(i in 1:length(rawPaths)){
```

Comment:
It's generally better practice to use `seq_along(rawPaths)`, rather than `1:length(x)` (which I also do all the time). It allows the code to fail more gracefully.  See the discussion here: https://stackoverflow.com/questions/24917228/proper-way-to-loop-over-the-length-of-a-dataframe-in-r